src/awkward/nplike.py

```diff
@@ -458,13 +468,40 @@ def char(self):
     def ndarray(self):
         return self._module.ndarray
 
+    def raw(self, array, nplike):
+        if isinstance(nplike, Numpy):
+            return array
+        elif isinstance(nplike, Cupy):
+            cupy = Cupy.instance()
+            return cupy.asarray(array, dtype=array.dtype, order="C")
+        elif isinstance(nplike, ak._v2._typetracer.TypeTracer):
+            return ak._v2._typetracer.TypeTracerArray(
+                dtype=array.dtype, shape=array.shape
+            )
+        else:
+            raise TypeError(
+                "Invalid nplike, choose between nplike.Numpy, nplike.Cupy, Typetracer"
+            )
+
 
 class Cupy(NumpyLike):
     def to_rectilinear(self, array, *args, **kwargs):
         return ak.operations.convert.to_cupy(array, *args, **kwargs)
 
     def __getitem__(self, name_and_types):
-        raise NotImplementedError("no CUDA in v2 yet")
+        import sys
```

Comment:
Since `sys` is a standard library, there's no reason to not put its import up at the top of the file.